#!/usr/bin/env python3
PACKAGE = "val_soft_estop_monitor"

from dynamic_reconfigure.parameter_generator_catkin import *

# initialize parameter generator to define parameters
gen = ParameterGenerator()

# add parameters to generator
gen.add("debug",                bool_t,   0, "Debug for printing",                        False)
gen.add("joint_velocity_limit", double_t, 0, "Limit for velocity of any one joint",       5.0, 0.0, 5.0) # TODO default, min, max values?
gen.add("joint_torque_limit",   double_t, 0, "Limit for torque applied by any one joint", 5.0, 0.0, 5.0) # TODO default, min, max values?

# generate necessary files
exit(gen.generate(PACKAGE, "val_soft_estop_monitor", "JointMonitorParams"))

"""
See the following tutorials for how this cfg file was created:

General dynamic_reconfigure documentation:
http://wiki.ros.org/dynamic_reconfigure

dynamic_reconfigure tutorials:
http://wiki.ros.org/dynamic_reconfigure

Writing a cfg file:
http://wiki.ros.org/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile

dynamic_reconfigure for a C++ node:
http://wiki.ros.org/dynamic_reconfigure/Tutorials/SettingUpDynamicReconfigureForANode%28cpp%29
"""
